### PR TITLE
Web UI to enter hours

### PR DESCRIPTION
**Pull request by JamesReate**
this makes it easy to enter data from the web.

### CODE REVIEW

File: src/TimeTracker.Api/Controllers/WebReportController.cs [hunk 1]
```diff
@@ -23,15 +27,54 @@ public WebReportController(TimeTrackerDbContext dbContext)
...
+            {
+                await timeEntryService.CreateBillableTimeEntry(date, model.Hours, model.ProjectId.Value);
+            }
+            else
+            {
+                await timeEntryService.CreateNonBillableTimeEntry(date, model.Hours, model.NonbillReason,
+                    model.TimeEntryType);
+            }
+
+            return RedirectToAction("UserEntryReport");
```
Comment (ambercs) [inline src/TimeTracker.Api/Controllers/WebReportController.cs]: nice

File: src/TimeTracker.Api/Controllers/WebReportController.cs [hunk 2]
```diff
@@ -23,15 +27,54 @@ public WebReportController(TimeTrackerDbContext dbContext)
...
+        }
+
+        private async Task<Guid> GetUserId()
         {
             var ident = User.Identity as ClaimsIdentity;
             // refactor this line, also used in AccountController
```
Comment (ambercs) [inline src/TimeTracker.Api/Controllers/WebReportController.cs]: do we still need this comment?

File: src/TimeTracker.Api/Properties/launchSettings.json [hunk 3]
```diff
@@ -20,7 +20,7 @@
     "TimeTracker.Api": {
       "commandName": "Project",
       "launchBrowser": true,
-      "launchUrl": "/",
```
Comment (ambercs) [inline src/TimeTracker.Api/Properties/launchSettings.json]: was this the source of the double "/" lol

File: src/TimeTracker.Api/Views/WebReport/UserEntryReport.cshtml [hunk 4]
```diff
@@ -1,12 +1,50 @@
...
+    <div class="form-row">
+        <div class="col">
+            <input type="date" name="Date" class="form-control" value="@Model.Date.ToString("yyyy-MM-dd")" />
+        </div>
+        <div class="col">
+            @Html.DropDownListFor(m=> m.TimeEntryType, 
+                Html.GetEnumSelectList(typeof(TimeEntryTypeEnum)), 
+                htmlAttributes: new { onChange = "selectTimeEntry(this);", @class = "custom-select" })
+            
+            @* <select class="custom-select" name="TimeEntryType" onChange="selectTimeEntry(this);"> *@
```
Comment (ambercs) [inline src/TimeTracker.Api/Views/WebReport/UserEntryReport.cshtml]: can the commented out parts go?

File: src/TimeTracker.Library/Services/TimeEntryService.cs [hunk 5]
```diff
@@ -20,15 +20,13 @@ public TimeEntryService(in Guid userId, TimeTrackerDbContext db)
...
-        public async Task<Guid> CreateBillableTimeEntry(DateTime date, double hours, int billableClientId,
-            int projectId)
+        public async Task<Guid> CreateBillableTimeEntry(DateTime date, double hours, int projectId)
         {
             VerifyHoursBeforeAdding(date, hours);
 
             var model = new TimeEntry
             {
                 TimeEntryId = Guid.NewGuid(),
-                BillingClientId = billableClientId,
```
Comment (ambercs) [inline src/TimeTracker.Library/Services/TimeEntryService.cs]: any specific reason for taking out the billing client id?

File: src/TimeTracker.Api/Controllers/WebReportController.cs (hunk 2 above)
Comment (JamesReate) [reply src/TimeTracker.Api/Controllers/WebReportController.cs]: nope we do not, removing

File: src/TimeTracker.Api/Properties/launchSettings.json (hunk 3 above)
Comment (JamesReate) [reply src/TimeTracker.Api/Properties/launchSettings.json]: i thought it was! but looks like maybe not

File: src/TimeTracker.Api/Views/WebReport/UserEntryReport.cshtml (hunk 4 above)
Comment (JamesReate) [reply src/TimeTracker.Api/Views/WebReport/UserEntryReport.cshtml]: yup, good catch forgot to delete

File: src/TimeTracker.Library/Services/TimeEntryService.cs (hunk 5 above)
Comment (JamesReate) [reply src/TimeTracker.Library/Services/TimeEntryService.cs]: didn't make sense b/c already have access to the billingClient via the ProjectId, ie. projects belong to billing clients. Gonna remove from Model as well and do the db migration

**Pull request merged**